src/Core/Checkout/Customer/Aggregate/CustomerGroup/DataAbstractionLayer/CustomerGroupRepositoryDecorator.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Customer\Aggregate\CustomerGroup\DataAbstractionLayer;
+
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityDefinition;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Event\EntityWrittenContainerEvent;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\AggregationResult\AggregationResultCollection;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\IdSearchResult;
+use Shopware\Core\Framework\DataAbstractionLayer\Write\Validation\RestrictDeleteViolation;
+use Shopware\Core\Framework\DataAbstractionLayer\Write\Validation\RestrictDeleteViolationException;
+
+class CustomerGroupRepositoryDecorator implements EntityRepositoryInterface
```

Comment:
Pls Nooo. Don't override repositories.

Just listen to PreValidateEvent and throw an error

Reply:
I was not aware of this event. Is there a link in the docs explaining the EntityRepository events? :trollface:

Reply:
It's really low level, but should catch all updates - even those going directly through the writer. You can take a look at this example: https://github.com/shopware/platform/blob/c38c7b2b568ca430bb7231c998ef0fd140a43da4/src/Core/System/Language/LanguageValidator.php#L69-L94, if you want want to prevent a delete.

Reply:
I'll take a look at it.